validator/src/main/java/com/amazon/aoc/validators/TraceValidator.java

```diff
@@ -64,23 +73,35 @@ public void validate() throws Exception {
     List<String> traceIdList = Collections.singletonList(traceId);
 
     // get retrieved trace from x-ray service
-    Map<String, Object> retrievedTrace = this.getRetrievedTrace(traceIdList);
-    log.info("value of retrieved trace map: {}", retrievedTrace);
-    // data model validation of other fields of segment document
-    for (Map.Entry<String, Object> entry : storedTrace.entrySet()) {
-      String targetKey = entry.getKey();
-      if (retrievedTrace.get(targetKey) == null) {
-        log.error("mis target data: {}", targetKey);
-        throw new BaseException(ExceptionCode.DATA_MODEL_NOT_MATCHED);
-      }
-      if (!entry.getValue().toString().equalsIgnoreCase(retrievedTrace.get(targetKey).toString())) {
-        log.error("data model validation failed");
-        log.info("mis matched data model field list");
-        log.info("value of stored trace map: {}", entry.getValue());
-        log.info("value of retrieved map: {}", retrievedTrace.get(entry.getKey()));
-        log.info("==========================================");
-        throw new BaseException(ExceptionCode.DATA_MODEL_NOT_MATCHED);
-      }
+    boolean isMatched = RetryHelper.retry(10,
+        Integer.parseInt(GenericConstants.SLEEP_IN_MILLISECONDS.getVal()),
+        false,
+        () -> {
+        Map<String, Object> retrievedTrace = this.getRetrievedTrace(traceIdList);
+        log.info("value of retrieved trace map: {}", retrievedTrace);
+        // data model validation of other fields of segment document
+        for (Map.Entry<String, Object> entry : storedTrace.entrySet()) {
+          String targetKey = entry.getKey();
+          if (retrievedTrace.get(targetKey) == null) {
+            log.error("mis target data: {}", targetKey);
+            throw new BaseException(ExceptionCode.DATA_MODEL_NOT_MATCHED);
+          }
+          if (!entry
+                  .getValue()
+                  .toString()
+                  .equalsIgnoreCase(retrievedTrace.get(targetKey).toString())) {
```

Comment:
Can we support regex here? If not, we can remove `name` from template like serverlessrepo-aot-py38-sar/api, serverlessrepo-aot-py38-sar-function-XMKMYMQM3776

Reply:
Yeah I'm going to add regex support in a separate PR, didn't want to clutter this one too much. Will remove the `name` in the meantime.